src/qt/walletmodel.cpp

```diff
@@ -96,6 +102,38 @@ void WalletModel::updateStatus()
         emit encryptionStatusChanged(newEncryptionStatus);
 }
 
+void WalletModel::forceUpdateBalances()
+{
+    // boost::timer t;
+    // Get required locks upfront. This avoids the GUI from getting stuck on
+    // periodical polls if the core is holding the locks for a longer time -
+    // for example, during a wallet rescan.
+    TRY_LOCK(cs_main, lockMain);
```

Comment:
Hmm - interesting...  The original usage of forceUpdateBalances was actually previously in a timer, which would run every n seconds (7 IIRC) to force update the display because I still had some bits that weren't updating properly with signals.  Once that was resolved I disabled the forced update via timer so perhaps attempting to get locks is a duplicate - let me look into it